airflow/jobs.py

```diff
@@ -668,7 +657,28 @@ def _do_dags(self, dagbag, dags, tis_out):
             except Exception as e:
                 self.logger.exception(e)
 
+    @provide_session
+    def _garbage_collect(self, dag_run, session=None):
```

Comment:
Since garbage collection is often referred in the memory context, can we rename this? Maybe `_reset_state_for_orphaned_tasks`?